dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java

```diff
@@ -472,6 +473,22 @@ public boolean move(HTMLPageAsset page, Host host, Folder parent, User user)
         }
         return false;
     }
+
+    /**
+     * @see HTMLPageAssetAPI#findPagesByTemplate(Template, User, boolean)
+     */
+    @Override
+    public List<Contentlet> findPagesByTemplate(Template template, User user, boolean respectFrontendRoles)
+            throws DotDataException, DotSecurityException {
+
+        return permissionAPI.filterCollection(
+                    contentletAPI.search("+_all:" + template.getIdentifier() + " +baseType:"
+                                    + BaseContentType.HTMLPAGE.getType(), -1, 0, null, user,
+                            respectFrontendRoles),
+                    PermissionAPI.PERMISSION_READ,
+                    respectFrontendRoles,
+                    user);
+    }
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)